tools/src/main/java/com/example/core/broker/testkit/TopicStub.java

```diff
@@ -0,0 +1,50 @@
+package com.example.core.broker.testkit;
+
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import akka.actor.Props;
+import akka.stream.Materializer;
+import com.lightbend.lagom.javadsl.api.broker.Subscriber;
+import com.lightbend.lagom.javadsl.api.broker.Topic;
+
+/**
+ * Stubs a {@link Topic} to run tests without starting the complete broker infrastructure. While {@link TopicStub}
+ * should only be used inside the Stub's of your Service, you can distribute the {@link ProducerStub} returned by
+ * {@link TopicStub#producerStub()} so that clients (test code) can send messages via this stubbed topic.
+ *
+ * @param <T>
+ */
+public class TopicStub<T> implements Topic<T> {
+
+    private final Materializer materializer;
+    private final ActorRef bufferActor;
+    private String topicId;
+
+    TopicStub(String topicId, ActorSystem actorSystem, Materializer materializer) {
+        Props bufferProps = Props.create(TopicBufferActor.class);
+        bufferActor = actorSystem.actorOf(bufferProps);
+
+        this.materializer = materializer;
+        this.topicId = topicId;
+    }
+
+    @Override
+    public Topic.TopicId topicId() {
+        return TopicId.of(topicId);
+    }
+
+    @Override
+    public Subscriber<T> subscribe() {
+        return new SubscriberStub<>(topicId, bufferActor, materializer);
+    }
+
+    /**
+     * @return a {@link ProducerStub} that clients (test code) can use to send messages through this {@link TopicStub}.
+     */
+    public ProducerStub<T> producerStub() {
+        return (T message) -> {
+            bufferActor.tell(message, ActorRef.noSender());
+        };
```

Comment:
I think the point here is why do we need a `ProducerStub` at all?  Why not just have a `send` method?  At the moment you've got this boiler plate in every test:

```java
topicStub = topicFactory.forId(BidEvent.class, TOPIC_ID);
bidTopicStub = topicStub.producerStub();

...
bidTopicStub.send(...);
```

Why not just do:

```java
topicStub = topicFactory.forId(BidEvent.class, TOPIC_ID);

...
topicStub.send(...);
```

I think the reasons for making it this way originally now no longer exist.